src/java.base/share/classes/jdk/internal/classfile/impl/SplitConstantPool.java

```diff
@@ -489,9 +503,7 @@ public InterfaceMethodRefEntry interfaceMethodRefEntry(ClassEntry owner, NameAnd
 
     @Override
     public MethodTypeEntry methodTypeEntry(MethodTypeDesc descriptor) {
-        var ret = (AbstractPoolEntry.MethodTypeEntryImpl)methodTypeEntry(utf8Entry(descriptor.descriptorString()));
-        ret.sym = descriptor;
-        return ret;
+        return methodTypeEntry(utf8Entry(descriptor));
     }
```

Comment:
This method body can be moved to [`ConstantPoolBuilder​::methodTypeEntry​(MethodTypeDesc)`] and that method be made `default` like [`ConstantPoolBuilder​::classEntry​(ClassDesc)`], [`ConstantPoolBuilder​::packageEntry​(PackageDesc)`], and [`ConstantPoolBuilder​::moduleEntry​(ModuleDesc)`].

[`ConstantPoolBuilder​::methodTypeEntry​(MethodTypeDesc)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/classfile/constantpool/ConstantPoolBuilder.html#methodTypeEntry(java.lang.constant.MethodTypeDesc)
[`ConstantPoolBuilder​::classEntry​(ClassDesc)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/classfile/constantpool/ConstantPoolBuilder.html#classEntry(java.lang.constant.ClassDesc)
[`ConstantPoolBuilder​::packageEntry​(PackageDesc)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/classfile/constantpool/ConstantPoolBuilder.html#packageEntry(java.lang.constant.PackageDesc)
[`ConstantPoolBuilder​::moduleEntry​(ModuleDesc)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/classfile/constantpool/ConstantPoolBuilder.html#moduleEntry(java.lang.constant.ModuleDesc)

Reply:
too lazy to file a csr...